src/pages/node-express/lesson13-a1.mdx

```diff
@@ -0,0 +1,526 @@
+<form method="POST">
+  <div>
+    <label name="email">Enter your email:</label>
+    <input name="email" />
+  </div>
+  <div>
+    <label name="password">Enter your password:</label>
+    <input type="password" name="password" />
+  </div>
+  <div>
+    <button>Logon</button>
+    <a href="/">
+      <button type="button">Cancel</button>
+    </a>
+  </div>
+</form>
+<%- include("partials/footer.ejs") %>
+```
+
+`views/register.js`
+
+```html
+<%- include("partials/head.ejs") %> <%- include("partials/header.ejs") %>
+<form method="POST">
+  <div>
+    <label name="name">Enter your name:</label>
+    <input name="name" />
+  </div>
+  <div>
+    <label name="email">Enter your email:</label>
+    <input name="email" />
+  </div>
+  <div>
+    <label name="password">Enter your password:</label>
+    <input type="password" name="password" />
+  </div>
+  <div>
+    <label name="password1">Confirm your password:</label>
+    <input type="password" name="password1" />
+  </div>
+  <div>
+    <button>Register</button>
+    <a href="/">
+      <button type="button">Cancel</button>
+    </a>
+  </div>
+</form>
+<%- include("partials/footer.ejs") %>
+```
+
+Revised `views/partials/header.ejs`:
+
+```html
+<h1>The Jobs EJS Application</h1>
+<% if (user) { %>
+<p>User <%= user.name %> is logged on.</p>
+<form method="POST" action="/sessions/logoff">
+  <button>Logoff</button>
+</form>
+<% } %> <% if (errors) { errors.forEach((err) => { %>
+<div>Error: <%= err %></div>
+<% }) } %> <% if (info) { info.forEach((msg) => { %>
+<div>Info: <%= msg %></div>
+<% }) } %>
+<hr />
+```
+
+These changes won't suffice to do anything in the application, until routes are added to match.
+We need to follow best practices, with separate route and controller files.
+
+## Router and Controller
+
+Create a file `routes/sessionRoutes.js`, as follows:
+
+```js
+const express = require("express");
+const passport = require("passport");
+const router = express.Router();
+
+const {
+  logonShow,
+  registerShow,
+  registerDo,
+  logoff,
+} = require("../controllers/sessionController");
+
+router.route("/register").get(registerShow).post(registerDo);
+router
+  .route("/logon")
+  .get(logonShow)
+  .post(
+    passport.authenticate("local", {
+      successRedirect: "/",
+      failureRedirect: "/sessions/logon",
+      failureFlash: true,
+    }),
+    (req, res) => {
+      res.send("Not yet implemented.");
+    }
+  );
+router.route("/logoff").post(logoff);
+
+module.exports = router;
+```
+
+Ignore the passport lines for the moment. This just sets up the routes. We need to create
+a corresponding file `controllers/sessionController.js`. Here we use the `User` model. However,
+the file you copied makes some references to the JWT library. You must edit `models/User.js`
+to remove those references in order for `User.js` to load. We aren't using JWTs in this project.
+
+```js
+const User = require("../models/User");
+const parseVErr = require("../util/parseValidationErr");
```

Comment:
Using `camelCase` naming for this variable and the file instead of the existing assignment's `snake_case` just to keep things consistent with the rest of the naming patterns we've been using throughout these assignments